core/src/main/java/com/datastax/oss/driver/internal/core/metadata/MetadataManager.java

```diff
@@ -274,28 +290,26 @@ private SingleThreaded(InternalDriverContext context, DriverExecutionProfile con
       this.schemaParserFactory = context.getSchemaParserFactory();
     }
 
-    private void initNodes(
-        Set<InetSocketAddress> addresses, CompletableFuture<Void> initNodesFuture) {
-      apply(new InitContactPointsRefresh(addresses));
-      initNodesFuture.complete(null);
-    }
-
     private Void refreshNodes(Iterable<NodeInfo> nodeInfos) {
+      MetadataRefresh refresh =
+          didFirstNodeListRefresh
+              ? new FullNodeListRefresh(nodeInfos)
+              : new InitialNodeListRefresh(nodeInfos, contactPoints);
```

Comment:
looking at the naming I don't see a difference between `full` and `initial`. Maybe it should be called `PartialNodeListRefresh`?

Reply:
Yeah I struggled to find a good name. Both are full refreshes of the node list. `initial` is the first one, where we have to copy over contact points; `full` is any one after that, where we don't.